### PR TITLE
feat: Api contract for gift cards

### PR DESCRIPTION
**Pull request by Sangamesh26**
## Type of Change

- [x] New feature

## Description

Api contract for gift cards payment method

Api contract :

```
{
    "payment_method": "gift_card",
    "payment_method_type": "baby_gift_card",
    "payment_method_data": {
        "gift_card": {
            "number": "{gift_card_number}",
            "cvc": "{cvc}"
        }
    }
}
```

### Additional Changes

- [x] This PR modifies the API contract

## Motivation and Context

To add new payment method, gift cards

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -664,6 +664,18 @@ pub enum PaymentMethodData {
     MandatePayment,
     Reward(RewardData),
     Upi(UpiData),
+    GiftCard(Box<GiftCardData>),
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, serde::Deserialize, serde::Serialize, ToSchema)]
+#[serde(rename_all = "snake_case")]
```
Comment (SanchithHegde) [inline crates/api_models/src/payments.rs:671]: You don't need this annotation.

**Comment by SanchithHegde** [comment]
@Sangamesh26 Please address the failing CI checks.

**Review by SanchithHegde — APPROVED** [review]

**Comment by jagan-jaya** [comment]
@Sangamesh26  I think changes in VAS required for this. Please keep the VAS PR ready

**Review by jagan-jaya — APPROVED** [review]

**Comment by jagan-jaya** [comment]
Please add the testcase in https://hs-payments-test.netlify.app
@Sangamesh26

**Review by jagan-jaya — CHANGES_REQUESTED** [review]
@Sangamesh26 How reward and gift card differs? why don't we add it in the Reward type itself?

**Comment by Sangamesh26** [comment]
> @Sangamesh26 How reward and gift card differs? why don't we add it in the Reward type itself?

Reward type and gift card differs, customers have to buy gift cards and rewards are credited

**Review by Abhicodes-crypto — APPROVED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**